File: problem-003/p3.cc
#include<iostream>
#include <chrono>
#include<math.h>

bool IsPrime(long num);
long LargestFact(long num);



int main(){
  std::chrono::time_point<std::chrono::system_clock> start, end;
  start = std::chrono::system_clock::now();
  //main program
  std::cout << LargestFact(600851475143) << std::endl;
  //end of main program
  end   = std::chrono::system_clock::now();
  std::chrono::duration<double> elapsed_seconds = end-start;
  std::cout << elapsed_seconds.count() << " seconds" <<std::endl;
}


bool IsPrime(long num){
  if(num%2 ==0) return false;
  for(long i=sqrt(num); i>2; i--){
    if(num%i ==0) return false;
  }
  return true;
}

long LargestFact(long num){
  for(long i=sqrt(num); i>1; i--){
    if(num%i==0 && IsPrime(i)) return i;
  }
  return 1;
}

File: Python/problem-003/p3.py
import math
import time

start = time.time()

def findfact(num):
    current_max = 1
    for i in range(int(math.sqrt(num))):
        if i == 0:
            continue
        else:
            if num%i == 0 and isprime(i) == 'y':
                current_max = i
    return current_max


def isprime(num):
    counter = 0
    for i in range(int(math.sqrt(num))):
        if i == 0:
            continue
        else:
            if num%i == 0:
                counter = counter + 1
    if counter == 1:
        return 'y'
    else:
        return 'n'

print(findfact(600851475143))

run_time = time.time()-start
print(run_time)
